components/director/pkg/cronjob/cronjob.go

```diff
@@ -47,22 +47,25 @@ func (r *cronJobRunner) Start(ctx context.Context) {
 
 	for {
 		start := time.Now()
+		log.C(ctx).Infof("Starting CronJob %s execution", r.CronJob.Name)
 		r.CronJob.Fn(newCtx)
-		select {
-		case <-newCtx.Done():
+		if newCtx.Err() != nil {
+			log.C(ctx).Infof("CronJob %s stopped due to context done", r.CronJob.Name)
 			return
-		default:
-			jobTime := time.Since(start)
-			log.C(ctx).Infof("CronJob %s executed for %s", r.CronJob.Name, jobTime.String())
-			if jobTime < r.CronJob.SchedulePeriod {
-				waitPeriod := r.CronJob.SchedulePeriod - jobTime
-				log.C(ctx).Infof("Scheduling CronJob %s to run after %s", r.CronJob.Name, waitPeriod.String())
-
-				select {
-				case <-newCtx.Done():
-					return
-				case <-time.After(waitPeriod):
-				}
+		}
+		jobTime := time.Since(start)
```

Comment:
Isn't be better to not implement scheduling, but instead make a tiny script that is run as a cron job inside cluster?